src/infra/logging/logger.js

```diff
@@ -0,0 +1,113 @@
+const IO = require('crocks/IO');
+const getProp = require('crocks/Maybe/getProp');
+const isObject = require('crocks/predicates/isObject');
+const safe = require('crocks/Maybe/safe');
+const composeK = require('crocks/helpers/composeK');
+const maybeToArray = require('crocks/Maybe/maybeToArray');
+const chain = require('crocks/pointfree/chain');
+const objOf = require('crocks/helpers/objOf');
+
+const { Just } = require('crocks/Maybe');
+
+const {
+  createLogger,
+  format,
+  transports,
+} = require('winston');
+
+const {
+  combine,
+  timestamp,
+  label,
+  printf,
+  json,
+  colorize,
+} = format;
+
+const productionFormatter = (machineHost) => combine(
+  label({
+    label: machineHost,
+  }),
+  timestamp(),
+  json(),
+);
+
+const productionLogger = ({
+  machineHost,
+  slackWebHookChannel,
+  slackWebHookUrl,
+}) => createLogger({
+  format: productionFormatter(machineHost),
+  transports: [
+    new transports.Console(),
+    new transports.Slack({
+      level: 'warn',
+      channel: slackWebHookChannel,
+      webhook_url: slackWebHookUrl,
+      custom_formatter: (level, message, meta) => ({
+        text: `*${process.env.npm_package_name}*\n`
+          + `*${level.toUpperCase()}*: ${message}\n`
+          + `*TIME*: ${new Date()}\n`
+          + `*STACK*: ${meta.stack || '--'}`,
+      }),
+    }),
+  ],
+});
+
+const developmentFormatter = (machineHost) => combine(
+  timestamp(),
+  colorize(),
+  label({
+    label: machineHost,
+  }),
+  printf((i) => {
+    const { stack } = i;
+    const msg = JSON.stringify(i.message);
+    const prefix = `${i.timestamp} [${i.label}] ${i.level}`;
+    return stack ? `${prefix}: ${msg}\nSTACK: ${stack}` : `${prefix}: ${msg}`;
```

Comment:
Is this expected to work? I suppose `i` could be `null` or `undefined` when logging simple text messages, which would throw an uncaught `TypeError`;